schema.go

```diff
@@ -419,50 +436,69 @@ func (schema *Schema) ResolveSpaceIndex(s interface{}, i interface{}) (uint32, u
 		panic("unexpected type of space param")
 	}
 
-	if i != nil {
-		switch i := i.(type) {
-		case string:
-			if schema == nil {
-				return spaceNo, indexNo, fmt.Errorf("schema is not loaded")
-			}
-			if space == nil {
-				if space, ok = schema.SpacesById[spaceNo]; !ok {
-					return spaceNo, indexNo, fmt.Errorf("there is no space with id %d", spaceNo)
-				}
-			}
-			if index, ok = space.Indexes[i]; !ok {
-				err := fmt.Errorf("space %s has not index with name %s", space.Name, i)
-				return spaceNo, indexNo, err
-			}
-			indexNo = index.Id
-		case uint:
-			indexNo = uint32(i)
-		case uint64:
-			indexNo = uint32(i)
-		case uint32:
-			indexNo = i
-		case uint16:
-			indexNo = uint32(i)
-		case uint8:
-			indexNo = uint32(i)
-		case int:
-			indexNo = uint32(i)
-		case int64:
-			indexNo = uint32(i)
-		case int32:
-			indexNo = uint32(i)
-		case int16:
-			indexNo = uint32(i)
-		case int8:
-			indexNo = uint32(i)
-		case Index:
-			indexNo = i.Id
-		case *Index:
-			indexNo = i.Id
-		default:
-			panic("unexpected type of index param")
+	return spaceNo, nil
+}
+
+// ResolveIndex tries to resolve an index number.
+// Note: i can be a number, string, or an object of Index type.
+func (schema *Schema) ResolveIndex(i interface{}, spaceNo uint32) (uint32, error) {
+	var (
+		indexNo uint32
+		space   *Space
+		index   *Index
+		ok      bool
+	)
+
+	if i == nil {
+		return 0, nil
+	}
+	switch i := i.(type) {
+	case string:
+		if schema == nil {
+			return indexNo, fmt.Errorf("schema is not loaded")
+		}
+		if space, ok = schema.SpacesById[spaceNo]; !ok {
+			return indexNo, fmt.Errorf("there is no space with id %d", spaceNo)
+		}
+		if index, ok = space.Indexes[i]; !ok {
+			err := fmt.Errorf("space %s has not index with name %s", space.Name, i)
+			return indexNo, err
 		}
+		indexNo = index.Id
+	case uint:
+		indexNo = uint32(i)
+	case uint64:
+		indexNo = uint32(i)
+	case uint32:
+		indexNo = i
+	case uint16:
+		indexNo = uint32(i)
+	case uint8:
+		indexNo = uint32(i)
+	case int:
+		indexNo = uint32(i)
+	case int64:
+		indexNo = uint32(i)
+	case int32:
+		indexNo = uint32(i)
+	case int16:
+		indexNo = uint32(i)
+	case int8:
+		indexNo = uint32(i)
+	case Index:
+		indexNo = i.Id
+	case *Index:
+		indexNo = i.Id
+	default:
+		panic("unexpected type of index param")
 	}
 
-	return spaceNo, indexNo, nil
+	return indexNo, nil
+}
+
+func (schema *Schema) NamesUseSupported() bool {
+	if schema == nil {
```

Comment:
We could just create another implementation of the interface and get reed of the case.

Reply:
Updated. `Connection` struct now has new field `schemaResolver`, that is used instead of `Schema` for internal calls. By default it is set to `DefaultSchemaResolver`, that returns false on `NamesUseSupported` call.